src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -562,19 +562,18 @@ namespace Utils
             const auto& columnHandle = getColumnFamilyBasedOnName(columnName);
             if (columnHandle->GetName() != rocksdb::kDefaultColumnFamilyName)
             {
-                // Find the column handle on the list and drop it
-                const auto it =
-                    std::find_if(m_columnsInstances.begin(),
-                                 m_columnsInstances.end(),
-                                 [&columnName](const auto& handle) { return columnName == handle->GetName(); });
-
-                // Check if the column exists
-                if (it != m_columnsInstances.end())
-                {
-                    it->drop();
-                    m_columnsInstances.erase(it);
+                // Create an iterator for the current column family
```

Comment:
We are doing a batch operation if the column family is the default one, don't know why is the case but I tested this

```Cpp
        void deleteAll(const std::string& columnName)
        {
            const auto& columnHandle = getColumnFamilyBasedOnName(columnName);

            rocksdb::WriteBatch batch;
            std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions(), columnHandle.handle()));

            it->SeekToFirst();
            while (it->Valid())
            {
                batch.Delete(columnHandle.handle(), it->key());
                it->Next();
            }

            if (auto status = m_db->Write(rocksdb::WriteOptions(), &batch); !status.ok())
            {
                throw std::runtime_error("Error deleting data: " + status.ToString());
            }
        }
```

And its working as expected, please perform some benchmarks and tests to see if we can use batch operations in every situation

Reply:
Same case with the deleteAll without columnName

Reply:
Unfortunately, the impact on the method `deleteAll` without columnName is too big and requires deeper refactoring. After making this change


```cpp
while (it != m_columnsInstances.end())
            {
                if ((*it)->GetName() != rocksdb::kDefaultColumnFamilyName)
                {
                    const auto& columnHandle = getColumnFamilyBasedOnName((*it)->GetName());

                    rocksdb::WriteBatch batch;
                    std::unique_ptr<rocksdb::Iterator> itColumn(
                        m_db->NewIterator(rocksdb::ReadOptions(), columnHandle.handle()));

                    itColumn->SeekToFirst();
                    while (itColumn->Valid())
                    {
                        batch.Delete(columnHandle.handle(), itColumn->key());
                        itColumn->Next();
                    }

                    if (auto status = m_db->Write(rocksdb::WriteOptions(), &batch); !status.ok())
                    {
                        throw std::runtime_error("Error deleting data: " + status.ToString());
                    }
                    columnsNames.push_back((*it)->GetName());
                    it = m_columnsInstances.erase(it);
                }
                else
                {
                    rocksdb::WriteBatch batch;
                    std::unique_ptr<rocksdb::Iterator> itDefault(
                        m_db->NewIterator(rocksdb::ReadOptions(), it->handle()));

                    itDefault->SeekToFirst();
                    while (itDefault->Valid())
                    {
                        batch.Delete(it->handle(), itDefault->key());
                        itDefault->Next();
                    }

                    if (const auto status = m_db->Write(rocksdb::WriteOptions(), &batch); !status.ok())
                    {
                        throw std::runtime_error("Error deleting data: " + status.ToString());
                    }

                    ++it;
                }
            }
```

All tests failed

```bash
[ RUN      ] RocksDBWrapperTest.CreateColumnTwiceThrows
unknown file: Failure
C++ exception with description "Couldn't create column family: Column family already exists" thrown in the test body.
unknown file: Failure
C++ exception with description "Couldn't create column family: Column family already exists" thrown in TearDown().
[  FAILED  ] RocksDBWrapperTest.CreateColumnTwiceThrows (30 ms)
```

I keep the lines

```cpp
 columnsNames.push_back((*it)->GetName());
 it = m_columnsInstances.erase(it);
```

Because is used later on the column recreation, and iteration

Reply:
## Update
After a huddle, we discussed the following changes

Benchmark the deleteAll without columns as follows:
- Create a huge DB (emulating a real scenario) with 1.000.000 elements distributed between multiple columns
- Test the drop column vs batch delete approach

Reply:
All requested changes done